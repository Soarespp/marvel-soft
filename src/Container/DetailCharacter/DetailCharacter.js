//react
import React, { useEffect, useState } from 'react';
import { useParams } from "react-router-dom";

//Redux
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import * as actionsCharacters from '../../store/actions/character/index';


//stylles
import { ImgDefault } from '../../Components/imgComponent/style';
import { DetailContainer, ContainerImg } from './DetailCharacter.style';

//Imports
import Header from '../../Components/Header/index';
import { Form, Input, Button, Row, List } from 'antd';

const initChar = {
    id: 0,
    name: '',
    series: { items: [] }
}

const DetailCharacter = ({ characters, changeCharacter, history }) => {
    const [charLocal, setCharLocal] = useState(initChar);
    var params = useParams();

    useEffect(() => {
        if (parseInt(params.idCharacter) !== -1) {
            const characterEdit = characters.find(comp => comp.id === parseInt(params.idCharacter)) || initChar;
            setCharLocal(characterEdit);
        }
    }, [characters, params.idCharacter])

    const handleSubmit = (data) => {
        changeCharacter(data)
        alert('salvo com sucesso')
        history.push('/')
    }

    return (
        <div>
            <Header seacher={false} />
            <DetailContainer>
                <ContainerImg>
                    {(charLocal.thumbnail) ?
                        <ImgDefault width='100%' height='100%' src={`${charLocal.thumbnail.path}.${charLocal.thumbnail.extension}`} />
                        : null}
                </ContainerImg>
                <Form >
                    <Row>
                        <Form.Item label="Name">
                            <Input
                                name="name"
                                placeholder="Name"
                                testID="name"
                                value={charLocal.name}
                                onChange={({ target: { name, value } }) => setCharLocal((local) => ({ ...local, [name]: value }))}
                                size="large"
                            />
                        </Form.Item>
                    </Row>
                    <Row>
                        <Form.Item label="Descrição">
                            <Input
                                name="description"
                                placeholder="description"
                                value={charLocal.description}
                                onChange={({ target: { name, value } }) => setCharLocal((local) => ({ ...local, [name]: value }))}
                                size="large"
                            />
                        </Form.Item>
                    </Row>
                    <Row>
                        <List
                            size="small"
                            header={<div>Series</div>}
                            bordered
                            dataSource={charLocal.series.items}
                            renderItem={item => <List.Item>{item.name}</List.Item>}
                        />
                    </Row>
                    <Form.Item>
                        <Button
                            name="save"
                            type="primary"
                            onClick={() => handleSubmit(charLocal)}
                        >
                            Save
                        </Button>
                        <Button
                            name="cancel"
                            type="outline"
                            testID="button"
                            onClick={() => history.push('/')}
                        >
                            Cancel
                        </Button>
                    </Form.Item>
                </Form>
            </DetailContainer>
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        characters: state.dados.characters,
    };
}

const mapDispatchToProp = (dispatch) => bindActionCreators(actionsCharacters, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProp
)(DetailCharacter);

