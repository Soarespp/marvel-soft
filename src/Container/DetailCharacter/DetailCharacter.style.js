import styled from "styled-components";

export const DetailContainer = styled.div`
    grid-template-columns: repeat(auto-fill, minmax(300px, 45%));
     grid-template-rows: minmax(150px, 100%);
    // Height:200px;
    display: grid;
    margin: auto;
    max-width: 1200px;
    border: 3px groove rgb(191, 6, 3);
    border-radius: 3px;
    grid-gap: 5px;
`;

export const ContainerImg = styled.div`
    width: 100%;
    height: 100%;
    min-width: 150px;
    min-Height: 300px;
    align-items: center;
    align-content: center;
    text-align: center;
    margin: auto;
`;