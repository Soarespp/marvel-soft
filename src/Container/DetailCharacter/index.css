.DetailCharacter {}

.DetailCharacter .Container-Data {
    grid-template-columns: repeat(auto-fill, minmax(360px, 45%));
    grid-template-rows: minmax(150px, 100%);
    display: grid;
    margin: auto;
    max-width: 1200px;
    border: 3px groove rgb(191, 6, 3);
    border-radius: 3px;
    grid-gap: 5px;
}