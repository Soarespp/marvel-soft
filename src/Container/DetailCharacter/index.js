import React, { useEffect, useState } from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import * as actionsCharacters from '../../store/actions/character/index';
import { useParams } from "react-router-dom";
import Header from '../../Components/Header/index';
import './index.css';
import { ImgDefault } from '../../Components/imgComponent/style';

import { Form, Input, Button, Row, List } from 'antd';

const initChar = {
    id: 0,
    name: '',
    series: { items: [] }
}

const DetailCharacter = (props) => {
    const { characters, changeCharacter, history } = props;
    const [charLocal, setCharLocal] = useState(initChar);
    var params = useParams();

    useEffect(() => {
        if (parseInt(params.idCharacter) !== -1) {
            const characterEdit = characters.find(comp => comp.id === parseInt(params.idCharacter)) || initChar;
            setCharLocal(characterEdit);
        }
    }, [characters, params.idCharacter])

    function handleSubmit(data) {
        changeCharacter(data)
        alert('salvo com sucesso')
        history.push('/')
    }

    return (
        <div className='DetailCharacter'>
            <Header seacher={false} />
            <div className='Container-Data'>
                <div style={{ backgroundColor: 'red', width: '100%', height: '100%', minWidth: '200px', minHeight: '300px' }}>
                    {(charLocal.thumbnail) ?
                        <ImgDefault width='100%' height='100%' src={`${charLocal.thumbnail.path}.${charLocal.thumbnail.extension}`} />
                        : null}
                </div>
                <Form >
                    <Row>
                        <Form.Item label="Name">
                            <Input
                                name="name"
                                testID='editName'
                                placeholder="Name"
                                value={charLocal.name}
                                onChange={({ target: { name, value } }) => setCharLocal((local) => ({ ...local, [name]: value }))}
                                size="large"
                            />
                        </Form.Item>
                    </Row>
                    <Row>
                        <Form.Item label="Descrição">
                            <Input
                                name="description"
                                placeholder="description"
                                value={charLocal.description}
                                onChange={({ target: { name, value } }) => setCharLocal((local) => ({ ...local, [name]: value }))}
                                size="large"
                            />
                        </Form.Item>
                    </Row>
                    <Row>
                        <List
                            size="small"
                            header={<div>Series</div>}
                            bordered
                            dataSource={charLocal.series.items}
                            renderItem={item => <List.Item>{item.name}</List.Item>}
                        />
                    </Row>
                    <Form.Item>
                        <Button
                            name="save"
                            type="primary"
                            onClick={() => handleSubmit(charLocal)}
                        >
                            Save
                        </Button>
                        <Button
                            name="cancel"
                            type="outline"
                            testID="button"
                            onClick={() => history.push('/')}
                        >
                            Cancel
                        </Button>
                    </Form.Item>
                </Form>
            </div>
        </div>
    );
}

function mapStateToProps(state) {
    return {
        characters: state.dados.characters,
    };
}

const mapDispatchToProp = (dispatch) => bindActionCreators(actionsCharacters, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProp
)(DetailCharacter);

