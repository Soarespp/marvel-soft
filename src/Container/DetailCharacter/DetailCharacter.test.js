import DetailCharacter from "./DetailCharacter";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("<DetailCharacter />", () => {
    let props;
    const mockStore = configureMockStore([thunk])
    beforeEach(() => {
        props = {
            characters: [{ id: 1, name: 'teste' }],
            idCharacter: 1,
        }
    })

    const store = mockStore({
        dados: {
            filter: false,
            pageCount: 0,
            bkpCharactersFilter: [],
            characters: [],
        }
    });

    afterEach(() => {
        jest.clearAllMocks;
    })

    it('renders', () => {
        const wrapper = shallow(<Provider store={store}><DetailCharacter {...props} /></Provider>);
        expect(wrapper).toBeTruthy();
    });
});
