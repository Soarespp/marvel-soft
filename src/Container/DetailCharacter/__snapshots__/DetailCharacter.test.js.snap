// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCharacter /> render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Memo(Connect(DetailCharacter))
    characters={
      Array [
        Object {
          "id": 1,
          "name": "teste",
        },
      ]
    }
    idCharacter={1}
  />
</Context.Provider>
`;

exports[`<DetailCharacter /> should render cases container 1`] = `ShallowWrapper {}`;
